server/src/main/java/io/crate/expression/scalar/conditional/MultiCaseFunction.java

```diff
@@ -0,0 +1,105 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar.conditional;
+
+import static io.crate.metadata.functions.TypeVariableConstraint.typeVariable;
+import static io.crate.types.TypeSignature.parseTypeSignature;
+
+import java.util.List;
+
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.BoundSignature;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+
+public class MultiCaseFunction extends Scalar<Object, Object> {
+
+    public static void register(ScalarFunctionModule module) {
+        // multi_case([boolean] conditions, [E] results) -> E
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.BOOLEAN_ARRAY.getTypeSignature(),
+                parseTypeSignature("array(E)"),
+                parseTypeSignature("E")
+            ).withTypeVariableConstraints(typeVariable("E")),
+            MultiCaseFunction::new
+        );
+        // multi_case([boolean] conditions, [E] results, E default) -> E
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.BOOLEAN_ARRAY.getTypeSignature(),
+                parseTypeSignature("array(E)"),
+                parseTypeSignature("E"),
+                parseTypeSignature("E")
+            ).withTypeVariableConstraints(typeVariable("E")),
+            MultiCaseFunction::new
+        );
+    }
+
+    public static final String NAME = "multi_case";
+
+    private final Signature signature;
+    private final BoundSignature boundSignature;
+
+    private MultiCaseFunction(Signature signature, BoundSignature boundSignature) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+    }
+
+    @Override
+    public Signature signature() {
+        return signature;
+    }
+
+    @Override
+    public BoundSignature boundSignature() {
+        return boundSignature;
+    }
+
+    @Override
+    public Object evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Object>[] args) {
+        List<Boolean> conditions = (List<Boolean>) args[0].value();
+        List<Object> values = null;
+        for (int i = 0; i < conditions.size(); i++) {
+            if (conditions.get(i)) {
```

Comment:
if conditions.get(i) is null I guess it's ok, doesn't go inside the block, right?

Reply:
Good catch, thank you. This was broken. I provided a fix and test.